File: sum.java
// time java -Xint sum.java
public class Sum {
    public static void main(String[] args) {
        long s = 0;
        for (long i = 0; i < 100000000; i++) {
            s += i;
        }
        System.out.println(s);
    }
}

File: sum.c
// gcc sum.c -O2 && time ./a.out

#include <stdio.h>

void main() {
    long s = 0;
    for (long i = 0; i < 100000000; i++) {
        s += i;
        asm(""); // avoid gcc optimizing away the entire loop
    }
    printf("%ld\n", s);
}